mmdet3d/models/necks/view_transformer.py

```diff
@@ -0,0 +1,310 @@
+            voxel during the acceleration.
+    """
+
+    def __init__(self,
+                 grid_config,
+                 input_size,
+                 downsample,
+                 in_channels,
+                 tran_channels,
+                 accelerate=False,
+                 max_voxel_points=300):
+        super(ViewTransformerLiftSplatShoot, self).__init__()
+        self.grid_config = grid_config
+        self.gen_grid_infos(**self.grid_config)
+
+        self.input_size = input_size
+        self.downsample = downsample
+
+        self.create_frustum()
+        self.tran_channels = tran_channels
+        self.depthnet = nn.Conv2d(
+            in_channels, self.D + self.tran_channels, kernel_size=1, padding=0)
+        self.accelerate = accelerate
+        self.max_voxel_points = max_voxel_points
+        self.initial_flag = True
+
+    def gen_grid_infos(self, x, y, z, **kwargs):
+        """Generate the grid information including the lower bound, interval,
+        and size.
+
+        Args:
+            x: Config of grid alone x axis in format of (lower_bound,
+                upper_bound, interval).
+            y: Config of grid alone y axis in format of (lower_bound,
+                upper_bound, interval).
+            z: Config of grid alone z axis in format of (lower_bound,
+                upper_bound, interval).
+            **kwargs: Container for other potential parameters
+        """
+        self.grid_lower_bound = \
+            nn.Parameter(torch.Tensor([cfg[0] + cfg[2]/2.0
+                                       for cfg in [x, y, z]]),
+                         requires_grad=False)
+        self.grid_interval = \
+            nn.Parameter(torch.Tensor([cfg[2] for
+                                       cfg in [x, y, z]]),
+                         requires_grad=False)
+        self.grid_size = \
+            nn.Parameter(torch.Tensor([(cfg[1] - cfg[0]) / cfg[2]
+                                       for cfg in [x, y, z]]),
+                         requires_grad=False)
+
+    def create_frustum(self):
+        """Generate the frustum template for each image."""
+        H_in, W_in = self.input_size
+        H_feat, W_feat = H_in // self.downsample, W_in // self.downsample
+        d = torch.arange(*self.grid_config['depth'], dtype=torch.float)\
+            .view(-1, 1, 1).expand(-1, H_feat, W_feat)
+        self.D = d.shape[0]
+        x = torch.linspace(0, W_in - 1, W_feat,  dtype=torch.float)\
+            .view(1, 1, W_feat).expand(self.D, H_feat, W_feat)
+        y = torch.linspace(0, H_in - 1, H_feat,  dtype=torch.float)\
+            .view(1, H_feat, 1).expand(self.D, H_feat, W_feat)
+
+        # D x H x W x 3
+        frustum = torch.stack((x, y, d), -1)
+        self.frustum = nn.Parameter(frustum, requires_grad=False)
+
+    def get_lidar_coor(self, rots, trans, intrins, post_rots, post_trans):
+        """Calculate the locations of the frustum points in the lidar(ego)
```

Comment:
lidar or ego? They can be different for different cases while the function name is get_lidar_coor. would be better to have a deterministic function